src/CalculatorUnitTests/UnitConverterViewModelUnitTests.cpp

```diff
@@ -454,57 +454,189 @@ namespace CalculatorUnitTests
+                NumbersAndOperatorsEnum::BINPOS10,
+                NumbersAndOperatorsEnum::BINPOS11,
+                NumbersAndOperatorsEnum::BINPOS12,
+                NumbersAndOperatorsEnum::BINPOS13,
+                NumbersAndOperatorsEnum::BINPOS14,
+                NumbersAndOperatorsEnum::BINPOS15,
+                NumbersAndOperatorsEnum::BINPOS16,
+                NumbersAndOperatorsEnum::BINPOS17,
+                NumbersAndOperatorsEnum::BINPOS18,
+                NumbersAndOperatorsEnum::BINPOS19,
+                NumbersAndOperatorsEnum::BINPOS20,
+                NumbersAndOperatorsEnum::BINPOS21,
+                NumbersAndOperatorsEnum::BINPOS22,
+                NumbersAndOperatorsEnum::BINPOS23,
+                NumbersAndOperatorsEnum::BINPOS24,
+                NumbersAndOperatorsEnum::BINPOS25,
+                NumbersAndOperatorsEnum::BINPOS26,
+                NumbersAndOperatorsEnum::BINPOS27,
+                NumbersAndOperatorsEnum::BINPOS28,
+                NumbersAndOperatorsEnum::BINPOS29,
+                NumbersAndOperatorsEnum::BINPOS30,
+                NumbersAndOperatorsEnum::BINPOS31,
+                NumbersAndOperatorsEnum::BINPOS32,
+                NumbersAndOperatorsEnum::BINPOS33,
+                NumbersAndOperatorsEnum::BINPOS34,
+                NumbersAndOperatorsEnum::BINPOS35,
+                NumbersAndOperatorsEnum::BINPOS36,
+                NumbersAndOperatorsEnum::BINPOS37,
+                NumbersAndOperatorsEnum::BINPOS38,
+                NumbersAndOperatorsEnum::BINPOS39,
+                NumbersAndOperatorsEnum::BINPOS40,
+                NumbersAndOperatorsEnum::BINPOS41,
+                NumbersAndOperatorsEnum::BINPOS42,
+                NumbersAndOperatorsEnum::BINPOS43,
+                NumbersAndOperatorsEnum::BINPOS44,
+                NumbersAndOperatorsEnum::BINPOS45,
+                NumbersAndOperatorsEnum::BINPOS46,
+                NumbersAndOperatorsEnum::BINPOS47,
+                NumbersAndOperatorsEnum::BINPOS48,
+                NumbersAndOperatorsEnum::BINPOS49,
+                NumbersAndOperatorsEnum::BINPOS50,
+                NumbersAndOperatorsEnum::BINPOS51,
+                NumbersAndOperatorsEnum::BINPOS52,
+                NumbersAndOperatorsEnum::BINPOS53,
+                NumbersAndOperatorsEnum::BINPOS54,
+                NumbersAndOperatorsEnum::BINPOS55,
+                NumbersAndOperatorsEnum::BINPOS56,
+                NumbersAndOperatorsEnum::BINPOS57,
+                NumbersAndOperatorsEnum::BINPOS58,
+                NumbersAndOperatorsEnum::BINPOS59,
+                NumbersAndOperatorsEnum::BINPOS60,
+                NumbersAndOperatorsEnum::BINPOS61,
+                NumbersAndOperatorsEnum::BINPOS62,
+                NumbersAndOperatorsEnum::BINPOS63,
+                NumbersAndOperatorsEnum::BINEND,
+                NumbersAndOperatorsEnum::Hyp
+            };
+
+            for (const auto& button : buttons)
```

Comment:
So I'm not entirely convinced it's useful to make sure that we don't actually do anything when any of the above commands are sent.  I'm ok with just removing the loop entirely if this test doesn't make sense and just keep the part of this test that covers supported commands.

Reply:
Based on the observation in the original bug (#175) and the position of `Zero` and `None` in [`CalculatorButtonUser.h`](https://github.com/Microsoft/calculator/blob/057401f5f2b4bb1ea143da02c773ac18d1bb9a2e/src/CalcViewModel/Common/CalculatorButtonUser.h#L63), I don't believe that the _intention_ by the original author at the time was to test _all_ of these commands, but rather just the ones between `Add` and `None`.  I did check the archive, and it's been this way since the test was first introduced (at that time, `NumbersAndOperatorsEnum` ended after `Cube`, so there _were_ commands after `None`).

I think it _can_ be reasonable to have tests that validate that certain commands do nothing in certain modes, however I think they only make sense if it would be clear to the maintainers what the failing test would indicate.  So, if we had this test currently pass because `Add` does nothing, but then in some mythical future we added the ability to add measurements together which would cause this test to fail, what would our action item be?  We'd probably just remove `Add` from `buttons` and give it its own unique case.  So, did it add value?  Not substantially.  Would there be value for still running this loop, but only for the subset of buttons between `Add` and `None` vs all of them?  Probably not.

I'd say go ahead and remove `buttons` and the loop entirely.

Reply:
Cool.  I'll go ahead and remove the loop.